source/04_sdk/02_reference/01_client_module/07_contribution/index.rst

```diff
@@ -0,0 +1,154 @@
+    
+Create contribution with modification of composition using version uid and save contribution 
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+.. code-block:: java
+
+    Composition composition = [..]
+
+    // Retrieve composition version uid 
+    String versionUid = composition.getVersionUid().toString()
+
+    // Create audit details 
+    AuditDetails audit = [..]
+
+    // Create contribution with composition modification change type and version uid
+     ContributionCreateDto contribution = ContributionBuilder.builder(audit)
+                    .addCompositionModification(composition, versionUid)
+                    .build();
+    
+    // Save contribution 
+    VersionUid versionUid = openEhrClient.contributionEndpoint(ehr).saveContribution(contribution);
+
+
+Create contribution with deletion of composition save contribution 
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+.. code-block:: java
+
+    Composition composition = [..]
+
+    // Retrieve composition version uid 
+    String versionUid = composition.getVersionUid().toString()
+
+    // Create audit details 
+    AuditDetails audit = [..]
+
+    // Create contribution with composition deletion change type
+    ContributionCreateDto contribution = ContributionBuilder.builder(audit)
+                    .addCompositionDeletion(versionUid)
+                    .build();
+    
+    // Contribution Saved
+    VersionUid versionUid = openEhrClient.contributionEndpoint(ehr).saveContribution(contribution);
+
+Create contribution using ``ContributionCreateDto``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``ContributionCreateDto`` class can be created by providing mandatory audit details and original versions.
+
+.. note:: Please provide data to all fields, which are required as per openEHR Reference Model. 
+    Otherwise the backend server will reject the payload as invalid.
+
+    A very simplified example of creating ``ContributionCreateDto``:
+
+.. code-block:: java
+
+    Composition composition = [..]
+    
+    // Create audit details 
+    AuditDetails audit = [..]
+    
+    // OriginalVersion must contain AuditDetails and can contain compositions depends on change type
+    List<OriginalVersion> originalVersions = [..]
+    originalVersion.setData(composition);
+    originalVersion.setCommitAudit(audit);
+    // Version uid must be provided in case of composition deletion and modification
+    String versionUid = composition.getVersionUid().toString()
+    originalVersion.setPrecedingVersionUid(new ObjectVersionId(versionUid));
+    
+    // Create contribution dto
+    ContributionCreateDto contribution = new ContributionCreateDto();
+    contributionCreateDto.setAudit(audit);
+    contributionCreateDto.setVersions(originalVersions);
+    
+    // Save contribution
+    VersionUid versionUid = openEhrClient.contributionEndpoint(ehr).saveContribution(contribution);
+
+Find contribution
```

Comment:
Maybe put this block before `Save contribution`, so it is not out of sight?

Reply:
changed